packages/frontend-bo/src/stores/demande-sejour.js

```diff
@@ -0,0 +1,76 @@
+import { defineStore } from "pinia";
+import { logger, $fetchBackend } from "#imports";
+import { getMonth } from "date-fns/getMonth";
+
+const log = logger("stores/demande-sejour");
+
+export const useDemandeSejourStore = defineStore("demandeSejour", {
+  state: () => ({
+    demandes: [],
+    total: 0,
+  }),
+  getters: {
+    getById: (state) => (stateId) =>
+      state.demandes.find((d) => d.demandeSejourId === parseInt(stateId)),
+    saison: (state) => {
+      return (stateId) => {
+        const demande = state.demandes.find(
+          (d) => d.demandeSejourId === stateId,
+        );
+        if (demande?.dateDebut) {
+          const moisDebut = getMonth(demande.dateDebut);
+          if (moisDebut < 3) return "Hiver";
+          if (moisDebut < 6) return "Printemps";
+          if (moisDebut < 9) return "Eté";
+          if (moisDebut < 12) return "Automne";
+        }
+      };
+    },
+    organismeTitle: (state) => (stateId) => {
```

Comment:
je ne sais pas si un getter qui renvoie une fonction a du sens mais j'imagine que ça ne fait pas de mal non plus.
Quand tu appelles cette fonction, as-tu juste un id ou as-tu déjà tout l'objet ? Si tu as juste l'id, c'est ok, sinon, comme c'est juste un format, il ne faut pas re-parcourir le tableau.

Reply:
Ici, je em suis inspiré de la doc :) https://pinia.vuejs.org/core-concepts/getters.html

![image](https://github.com/SocialGouv/vao/assets/139745995/a9103f17-eef2-4ea9-bdc8-fc8e0a058b12)

Dans l'utilisation, c'est assez transparent, pinia te chache l'aspect return d'une fonction. Tu appelle le getter directement avec le param : 

```
(value) => sejourStore.organismeTitle(value.demandeSejourId)
```

pareil, j'ai recopié la doc.

A priori, c'est la facon standart de declarer un getter avec parametre. Je sais pas si je reponds a tes question :)

Reply:
ce n'est pas vraiment le fait d'avoir une fonction qui renvoie une fonction dans le store qui me pose problème. 
Si tu lis un peu plus bas, tu as ceci : 
Note that when doing this, getters are not cached anymore, they are simply functions that you invoke.
Mettre ça dans le store permet de regrouper les fonctions, donc pourquoi pas.

En revanche, parcourir plusieurs fois un tableau dans plusieurs fonctions pour retrouver le même objet, ce n'est pas une bonne pratique. Il vaut mieux trouver l'objet dans ton tableau et le passer en paramètre de plusieurs fonctions. Dans notre cas, ça n'aura pas un impact énorme vu que le tableau est limité en taille a priori mais ce n'est pas une bonne pratique en géréral.

Plutôt que : 
```
function a(id) {
  return this.users.find(u => u.id == id).a
}

function b(id) {
  return this.users.find(u => u.id == id).b
}

console(a(1), b(1));
```

il vaut mieux avoir : 

```
function u(id) {
   return this.users.find(u => u.id == id)
}

function a(user) {
  return user.a
}

function b(user) {
  return user.b
}

var u1 = u(1);
console(a(u1), b(u1));
```

Je ne sais pas si c'est clair et cela dépend du contexte d'appel.